src/modules/mc_pos_control/MulticopterPositionControl.cpp

```diff
@@ -88,6 +100,35 @@ int MulticopterPositionControl::parameters_update(bool force)
 		ModuleParams::updateParams();
 		SuperBlock::updateParams();
 
+		if (_param_sys_vehicle_resp.get() >= 0.f) {
+			// make it less sensitive at the lower end
+			float responsiveness = _param_sys_vehicle_resp.get() * _param_sys_vehicle_resp.get();
+
+			int num_changed = 0;
+			num_changed += paramSet(_param_mpc_acc_hor, math::lerp(1.f, 15.f, responsiveness));
+			num_changed += paramSet(_param_mpc_acc_hor_max, math::lerp(1.f, 15.f, responsiveness));
+			num_changed += paramSet(_param_mpc_man_y_max, math::lerp(50.f, 300.f, responsiveness));
+
+			if (responsiveness > 0.6f) {
+				num_changed += paramSet(_param_mpc_man_y_tau, 0.f);
+
+			} else {
+				num_changed += paramSet(_param_mpc_man_y_tau, math::lerp(2.f, 0.f, responsiveness));
+			}
+
+			if (responsiveness < 0.5f) {
+				num_changed += paramSet(_param_mpc_tiltmax_air, 45.f);
+
+			} else {
+				num_changed += paramSet(_param_mpc_tiltmax_air, math::min(MAX_SAFE_TILT_DEG, math::lerp(45.f, 70.f,
```

Comment:
Before revising the maximum thrust limit to react really nicely we have to be careful with encourageing tilt limit increase. The current implementation is focused on altitude priorization for safety but can lead to unwanted quick tilt changes in case the maximum thrust is exceeded. This is especially a problem for under-powered vehicles.

Reply:
Yes I'm aware and only increase it for high values. With 0.8 it's still 52 deg, and above the vehicle is generally really responsive - I'll add a warning in QGC.